### DISCUSSION TITLE
4.8.0-0.okd-2021-11-14-052418 on aws fails to install

### DISCUSSION THREAD

**Discussion by jlatzko** [okd-project/okd#1218] [Q&A]
Hi!
This is my first try to install OKD into AWS (I need OKD 4.8 for compatibility reasons). I am also quite new to kubernetes...

I followed https://docs.okd.io/latest/installing/installing_aws/installing-aws-customizations.html

The only customization I made was to limit the control and compute plane to 3 availability zones to work around a problem with EIP resource limitations.
Part of install-config.yaml with zones added:

==
apiVersion: v1
baseDomain: at.nttdata-labs.com
compute:
- architecture: amd64
  hyperthreading: Enabled
  name: worker
  platform:
    aws:
      zones:
      - us-west-2a
      - us-west-2b
      - us-west-2c
  replicas: 3
controlPlane:
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform:
    aws:
      zones:
      - us-west-2a
      - us-west-2b
      - us-west-2c
  replicas: 3
===

I run the installer on a centos 8 machine.

Installer waits for the Kubernetes API and eventually fails:

time="2022-05-10T16:38:08+02:00" level=debug msg="Apply complete! Resources: 134 added, 0 changed, 0 destroyed."
time="2022-05-10T16:38:08+02:00" level=debug msg="OpenShift Installer 4.8.0-0.okd-2021-11-14-052418"
time="2022-05-10T16:38:08+02:00" level=debug msg="Built from commit acbb3f8204ed84fc25817a8a8957f97d13868d68"
time="2022-05-10T16:38:08+02:00" level=info msg="Waiting up to 20m0s for the Kubernetes API at https://api.okd-ntt.at.nttdata-labs.com:6443..."
time="2022-05-10T16:38:22+02:00" level=debug msg="Still waiting for the Kubernetes API: an error on the server (\"\") has prevented the request from succeeding"
time="2022-05-10T16:42:24+02:00" level=debug msg="Still waiting for the Kubernetes API: an error on the server (\"\") has prevented the request from succeeding"
time="2022-05-10T16:46:26+02:00" level=debug msg="Still waiting for the Kubernetes API: an error on the server (\"\") has prevented the request from succeeding"
time="2022-05-10T16:50:28+02:00" level=debug msg="Still waiting for the Kubernetes API: an error on the server (\"\") has prevented the request from succeeding"
time="2022-05-10T16:54:30+02:00" level=debug msg="Still waiting for the Kubernetes API: an error on the server (\"\") has prevented the request from succeeding"

Also in .openshift-install.log:
time="2022-05-10T16:30:14+02:00" level=debug msg="    Fetching Bootstrap SSH Key Pair..."
time="2022-05-10T16:30:14+02:00" level=debug msg="    Generating Bootstrap SSH Key Pair..."
time="2022-05-10T16:30:14+02:00" level=debug msg="Failed to marshal PKIX public key: %!s(<nil>)"

Is this where the problem starts?

I am able to attach a terminal to the bootstrap machine. No pods are running.
[core@ip-10-0-9-61 ~]$ sudo crictl ps -a
CONTAINER           IMAGE                                                              CREATED             STATE               NAME                ATTEMPT             POD ID
47eacf65d7489       1e07aceef7bc1c2336aaaa8f5d6bafe5baf099b0368d8f950738f8a69870d515   14 seconds ago      Exited              etcd                18                  4f6d0a6cca72e

I am stuck here. Where to look next?

Thanks,
Joachim.

**Comment by vrutkovs** [comment]
Bootstrap has completed but console pod can't resolve itself:
```
May 10 14:58:30 ip-10-0-132-114 hyperkube[1558]: I0510 14:58:30.511178    1558 prober.go:116] "Probe failed" probeType="Readiness" pod="openshift-console/console-7cfd458949-cz2ll" podUID=49dec64c-7c91-42fd-94bd-47f270aa405e containerName="console" probeResult=failure output="Get \"https://10.128.0.38:8443/health\": dial tcp 10.128.0.38:8443: connect: connection refused"
```

Check network configuration

**Reply by jlatzko** [reply]
I am sorry for asking these noobish question... What makes you think the bootstrap completed successfully?
I am still working my way through https://docs.okd.io/4.8/support/troubleshooting/troubleshooting-installations.html.
bootstrap node is still there, so the problem happened before the boostrap shutdown (9.)
Thee control plane nodes have been created and started (as expected, one for each availability zone)

Where did the installation fail? At stage 3 "The control plane machines fetch the remote resources from the bootstrap machine and finish booting."?

Network has been configured by the installer. It created all the ressources in AWS, I believe (3 load balancers etc.)

**Comment by fortinj66** [comment]
On the bootstrap look at the journal. There should be an example when you log into it

**Reply by jlatzko** [reply]
Hi! Not sure what you mean by "There should be an example when you log into it".
journalctl -b -f -u release-image.service -u bootkube.service
says things like 
May 10 14:47:58 ip-10-0-9-61 bootkube.sh[1531]: bootkube.service complete
and
May 10 14:47:59 ip-10-0-9-61 systemd[1]: bootkube.service: Deactivated successfully.
and
May 10 14:36:20 ip-10-0-9-61 systemd[1]: Finished Download the OpenShift Release Image.

That looks fine, correct?

**Comment by fortinj66** [comment]
Can you post the entire log from that?

**Reply by fortinj66** [reply]
Never mind.  I didn't see the log bundle above...  Disadvantage of looking from my phone

**Comment by fortinj66** [comment]
what does `oc get co` show?

**Reply by fortinj66** [reply]
and `oc get nodes -o wide`

**Reply by jlatzko** [reply]
both commands run on the bootstrap node:
# oc get co
NAME                                       VERSION                         AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.8.0-0.okd-2021-11-14-052418   False       True          False      14m
baremetal                                  4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
cloud-controller-manager                   4.8.0-0.okd-2021-11-14-052418   True        False         False      14m
cloud-credential                           4.8.0-0.okd-2021-11-14-052418   True        False         False      20m
cluster-autoscaler                         4.8.0-0.okd-2021-11-14-052418   True        False         False      11m
config-operator                            4.8.0-0.okd-2021-11-14-052418   True        False         False      14m
console                                    4.8.0-0.okd-2021-11-14-052418   False       True          False      8m40s
csi-snapshot-controller                    4.8.0-0.okd-2021-11-14-052418   True        False         False      13m
dns                                        4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
etcd                                       4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
image-registry                             4.8.0-0.okd-2021-11-14-052418   True        False         False      4m56s
ingress                                    4.8.0-0.okd-2021-11-14-052418   True        False         False      4m17s
insights                                   4.8.0-0.okd-2021-11-14-052418   True        False         False      6m57s
kube-apiserver                             4.8.0-0.okd-2021-11-14-052418   True        True          False      10m
kube-controller-manager                    4.8.0-0.okd-2021-11-14-052418   True        False         False      10m
kube-scheduler                             4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
kube-storage-version-migrator              4.8.0-0.okd-2021-11-14-052418   True        False         False      14m
machine-api                                4.8.0-0.okd-2021-11-14-052418   True        False         False      9m8s
machine-approver                           4.8.0-0.okd-2021-11-14-052418   True        False         False      13m
machine-config                             4.8.0-0.okd-2021-11-14-052418   True        False         False      11m
marketplace                                4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
monitoring                                                                 Unknown     True          Unknown    14m
network                                    4.8.0-0.okd-2021-11-14-052418   True        False         False      14m
node-tuning                                4.8.0-0.okd-2021-11-14-052418   True        False         False      5m17s
openshift-apiserver                        4.8.0-0.okd-2021-11-14-052418   True        False         False      8m7s
openshift-controller-manager               4.8.0-0.okd-2021-11-14-052418   True        False         False      11m
openshift-samples                          4.8.0-0.okd-2021-11-14-052418   True        False         False      7m53s
operator-lifecycle-manager                 4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
operator-lifecycle-manager-catalog         4.8.0-0.okd-2021-11-14-052418   True        False         False      12m
operator-lifecycle-manager-packageserver   4.8.0-0.okd-2021-11-14-052418   True        False         False      8m6s
service-ca                                 4.8.0-0.okd-2021-11-14-052418   True        False         False      14m
storage                                    4.8.0-0.okd-2021-11-14-052418   True        False         False      13m
# oc get nodes -o wide
NAME                                         STATUS   ROLES    AGE     VERSION                INTERNAL-IP    EXTERNAL-IP   OS-IMAGE           KERNEL-VERSION            CONTAINER-RUNTIME
ip-10-0-129-199.us-west-2.compute.internal   Ready    worker   8m7s    v1.21.2+9e8f924-1555   10.0.129.199   <none>        Fedora CoreOS 34   5.14.14-200.fc34.x86_64   cri-o://1.21.4
ip-10-0-157-235.us-west-2.compute.internal   Ready    master   18m     v1.21.2+9e8f924-1555   10.0.157.235   <none>        Fedora CoreOS 34   5.14.14-200.fc34.x86_64   cri-o://1.21.4
ip-10-0-182-219.us-west-2.compute.internal   Ready    master   17m     v1.21.2+9e8f924-1555   10.0.182.219   <none>        Fedora CoreOS 34   5.14.14-200.fc34.x86_64   cri-o://1.21.4
ip-10-0-185-5.us-west-2.compute.internal     Ready    worker   7m41s   v1.21.2+9e8f924-1555   10.0.185.5     <none>        Fedora CoreOS 34   5.14.14-200.fc34.x86_64   cri-o://1.21.4
ip-10-0-215-149.us-west-2.compute.internal   Ready    master   18m     v1.21.2+9e8f924-1555   10.0.215.149   <none>        Fedora CoreOS 34   5.14.14-200.fc34.x86_64   cri-o://1.21.4
ip-10-0-220-101.us-west-2.compute.internal   Ready    worker   8m9s    v1.21.2+9e8f924-1555   10.0.220.101   <none>        Fedora CoreOS 34   5.14.14-200.fc34.x86_64   cri-o://1.21.4

**Reply by jlatzko** [reply]
Please note: i destroyed the cluster, deleted the installation directory and created the cluster again. IP adresses from above do not match with the previously posted bundle log.

**Reply by jlatzko** [reply]
that is odd. wanted to paste a screenshot of the output so it is better readable and now the console pod shows as "available"


![image](https://user-images.githubusercontent.com/87323321/167742212-8d31dadf-6acf-4650-a4a2-f24e97a06ada.png)

**Reply by fortinj66** [reply]
I think you are OK then

**Reply by jlatzko** [reply]
The installer failed because the api server was not reachable from the machine the installer was running on. Looks like our corporate firewall is blocking certain requests or something like that.
oc did not work on that machine either. Everything worked fine from a machine outside our network.

Thanks!
